client/modules/IDE/components/Searchbar/Searchbar.jsx

```diff
@@ -1,71 +1,64 @@
+import React, { useState, useEffect, useCallback } from 'react';
 import PropTypes from 'prop-types';
-import React from 'react';
 import { throttle } from 'lodash';
 import { withTranslation } from 'react-i18next';
 import i18next from 'i18next';
 import SearchIcon from '../../../../images/magnifyingglass.svg';
 
-class Searchbar extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      searchValue: this.props.searchTerm
-    };
-    this.throttledSearchChange = throttle(this.searchChange, 500);
-  }
-
-  componentWillUnmount() {
-    this.props.resetSearchTerm();
-  }
-
-  handleResetSearch = () => {
-    this.setState({ searchValue: '' }, () => {
-      this.props.resetSearchTerm();
-    });
+const Searchbar = ({
+  searchTerm,
+  setSearchTerm,
+  resetSearchTerm,
+  searchLabel,
+  t
+}) => {
+  const [searchValue, setSearchValue] = useState(searchTerm);
+  const searchChange = useCallback(
+    throttle(() => setSearchTerm(searchValue.trim()), 500),
```

Comment:
I suspect that `throttle` doesn't do anything here because of the dependency on `searchValue`.

We have a bunch of PRs for this component already.  We need to look through and pick one that handles the throttle situation properly.